include/deal.II/lac/read_write_vector.templates.h

```diff
@@ -124,6 +124,19 @@ namespace LinearAlgebra
 
 
 
+#ifdef DEAL_II_WITH_CXX11
+  template <typename Number>
+  template <typename Functor>
+  void
+  ReadWriteVector<Number>::apply(const Functor &func)
+  {
+    FunctorTemplate<Functor> functor(*this, func);
+    internal::parallel_for(functor, n_elements(), thread_loop_partitioner);
```

Comment:
this is the place I was referring to -- `functor` has a lifetime shorter than the function argument, so storing a const reference would be ok.